### ISSUE TITLE
Trim whitespace from metadata field `docs`

### ISSUE THREAD

**Issue by cmichi** [paritytech/ink#920]
I thought we fixed this a long time ago, but apparently the field `docs` in our metadata still contains whitespaces at the beginning:

```bash
$ cd examples/flipper/
$ cargo +nightly contract build
$ cat /target/ink/flipper.contract  | jq .spec.messages[].docs
[
  " Flips the current value of the Flipper's boolean."
]
[
  " Returns the current value of the Flipper's boolean."
]
```

**Comment by Robbepop** [comment]
I am not sure if this is entirely bad since this is an artifact of the actual Rust code itself.
For example:
```rust
/// Foo.
fn foo() {}
```
Will produce `" Foo."` as captured documentation string whereas ...
```rust
///Foo.
fn foo() {}
```
Will produce `"Foo.".

Also if we treat the output as Markdown we should be careful with removal of whitespace since Markdown has semantic whitespace. E.g. indentation of a block is a code block.

**Comment by cmichi** [comment]
>I am not sure if this is entirely bad since this is an artifact of the actual Rust code itself.

Yeah sure, but I think we shouldn't require people to write comments as `///Foo.` just to get rid of whitespaces. Also `rustfmt` automatically corrects to `/// Foo.`. 

>Also if we treat the output as Markdown we should be careful with removal of whitespace since Markdown has semantic whitespace. E.g. indentation of a block is a code block.

I agree. I think a valid condition would be to only remove the whitespace if every entry in the `docs` Array starts with a whitespace.

**Comment by Robbepop** [comment]
> I agree. I think a valid condition would be to only remove the whitespace if every item in the docs Array starts with a whitespace.

Yes but just remove the general whitespace indentation because this is valid markdown, too:

```
The following lines are showing text as code:

    fn main() {
        "Hello, World!"
    }
```
However, if we see somthing like this:
```
 Every line stars with a single whitespace.
 So we can actually remove this single whitespace probably without interfering
 with the semantics of the whitespace below.
 
 The following lines are showing text as code:

    fn main() {
        "Hello, World!"
    }
```

**Comment by SkymanOne** [comment]
Fixed by #1385
